<launch>
	<arg name="veh" default="$(env VEHICLE_NAME)"/>
	<arg name="ai_trafo_mode" default="cb" doc="'cb' for colo balance only; 'both' for color balance and linear trafo"/>
	<arg name="ai_interval" default="5" doc="interval with which the linear trafo gets updated. color balance is performed every second."/>
	<arg name="verbose" default="false"/>
	<arg name="param_file_name" default="default" />
	<arg name="/my_lane_following/lane_controller" default="true" />
	<!-- start basic args -->
	<include file="$(find duckietown_demos)/launch/master.launch">
		<arg name="veh" value="$(arg veh)"/>
		<arg name="visualization" default="true" />
		<!--arg name="verbose" value="$(arg verbose)" /-->
    	<arg name="/camera/raw" default="true" />
		<arg name="param_file_name" default="default" />
		<arg name="line_detector_param_file_name" default="default" />
		<arg name="anti_instagram" default="true" />
		<arg name="/lane_following/lane_controller" default="false" />
		<arg name="ai_trafo_mode" value="$(arg ai_trafo_mode)"/>
		<arg name="ai_interval" value="$(arg ai_interval)"/>
		<arg name="/lane_following/stop_line_filter" default="false" /> <!-- will not every allow a transition into COORDINATION mode -->
		<arg name="vehicle_avoidance" default="false"/>
	</include>

	<!-- Lane controller -->
	<group if="$(arg /my_lane_following/lane_controller)">

		<!--remap from="lane_controller_node/seglist_filtered" to="lane_filter_node/seglist_filtered"/>
		<remap from="lane_controller_node/car_cmd" to="car_cmd_switch_node/car_cmd" /-->

		<include file="$(find pure-pursuit)/launch/lane_controller_node.launch">
			<arg name="veh" value="$(arg veh)"/>
			<arg name="param_file_name" value="$(arg param_file_name)"/>
		</include>
	</group>
</launch>
